tests/Doctrine/Tests/DBAL/Schema/SqliteSchemaDiffTest.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Schema;
+
+use Doctrine\DBAL\DBALException;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Schema\ForeignKeyConstraint;
+use Doctrine\DBAL\Schema\SchemaDiff;
+use Doctrine\DBAL\Schema\Table;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class SqliteSchemaDiffTest extends TestCase
```

Comment:
Are the tests specific to Sqlite or could they apply to other platforms?

Reply:
I'm not sure. They are testing a specific use case that I encountered with SQLite, it might apply to any platform where `supportsForeignKeyConstraints()` is `true` and `supportsCreateDropForeignKeyConstraints()` is `false`.

Reply:
Given the number of things mocked, I can barely understand what is being tested here. Would it make sense to have an integration test instead?

Reply:
I guess so. I just copied the code that was in the SchemaDiffTest. Maybe a more generic integration test could cover both cases.